src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterface.cs

```diff
@@ -1791,10 +1791,16 @@ private string ReadLineFromConsole(bool endOnTab, string initialContent, bool ca
                     }
 
                     // Modify string
-                    if (!insertMode) // then overwrite mode
+                    if (!insertMode && s != "") // then overwrite mode
                     {
                         s = s.Remove(index, 1);
```

Comment:
When it's the overwrite mode, `index` could be out of range here too, right?
And, if we check `index` here, it seems no reason to keep `&& s != ""`.

Reply:
Yes, move check up